### PR TITLE
Add step to gather dumps for any process launched from working folder.

### PR DESCRIPTION
**Pull request by Alan-Jowett**
## Description

Gather a crash dump if the test ends and the exe is still running.

## Testing

CI/CD

## Documentation

No.

### CODE REVIEW

**Review by dthaler — APPROVED** [review]

**Comment by Alan-Jowett** [comment]
Still running a test in my repo to see if the process gets killed in timeout. I.e. is there anything to gather a dump on?

**Review by dthaler — DISMISSED** [review]

File: scripts/Run-Test.ps1 [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Copyright (c) Microsoft Corporation
+# SPDX-License-Identifier: MIT
+
+# This script executes the provided test command, waits for <timeout in seconds>
+# and then captures a dump of the test process if it is still running. The dump
+# is captured using the procdump tool from Sysinternals. The dump is saved to
+# the <output folder> with the name of the test executable and the current date
+# and time.
+
+if ($args.Count -eq 0) {
```
Comment (saxena-anurag) [inline scripts/Run-Test.ps1:10]: We can define the arguments to this script using `param(...)`.

Example:

```
param ([switch]$Uninstall,
       [parameter(Mandatory=$false)][string] $LogFileName = "TestLog.log")
```

File: .github/workflows/reusable-test.yml [hunk 2]
```diff
@@ -198,7 +208,7 @@ jobs:
       shell: cmd
       run: |
           set EBPF_ENABLE_WER_REPORT=yes
-          OpenCppCoverage.exe -q --sources %CD% --excluded_sources %CD%\external\Catch2 --export_type cobertura:ebpf_for_windows.xml --working_dir ${{env.BUILD_PLATFORM}}\${{env.BUILD_CONFIGURATION}} -- ${{env.BUILD_PLATFORM}}\${{env.BUILD_CONFIGURATION}}\${{env.TEST_COMMAND}}
+          OpenCppCoverage.exe -q --sources %CD% --excluded_sources %CD%\external\Catch2 --export_type cobertura:ebpf_for_windows.xml --working_dir ${{env.BUILD_PLATFORM}}\${{env.BUILD_CONFIGURATION}} -- powershell .\Run-Test.ps1 ${{env.DUMP_PATH}} ${{env.TEST_TIMEOUT}} ${{env.TEST_COMMAND}}
 
     - name: Run test without Code Coverage
```
Comment (saxena-anurag) [inline .github/workflows/reusable-test.yml:213]: nit: should we use the same script in this case also? (`Run test without Code Coverage`)?

**Review by saxena-anurag — DISMISSED** [review]

File: scripts/Run-Test.ps1 [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+$args = $args[1..($args.Length - 1)]
+$Timeout = [int]$args[0]
+$args = $args[1..($args.Length - 1)]
+
+$process = Start-Process -PassThru -NoNewWindow -FilePath $args[0] -ArgumentList $args[1..($args.Length - 1)]
+
+if (!$process.WaitForExit($Timeout * 1000)) {
+    $dumpFileName = "$($process.ProcessName)_$(Get-Date -Format 'yyyy-MM-dd_HH-mm-ss').dmp"
+    $dumpFilePath = Join-Path $OutputFolder $dumpFileName
+    Write-Host "Capturing dump of $($process.ProcessName) to $dumpFilePath"
```
Comment (saxena-anurag) [inline scripts/Run-Test.ps1:25]: Should we use `Write-Output` instead? (here and other places)

File: .github/workflows/reusable-test.yml (hunk 2 above)
Comment (Alan-Jowett) [reply .github/workflows/reusable-test.yml:214]: That option should be renamed. It is for cases where we are running the tests inside the self-hosted runner.

Comment (saxena-anurag) [reply .github/workflows/reusable-test.yml:214]: Yeah I realized that later.

**Review by dthaler — APPROVED** [review]

**Pull request merged**